### PR TITLE
Fix test suite

### PR DESCRIPTION
**Pull request by penguincoder**
This fixes the test suite and remedies a small bug with `stop_worker` from stopping too many workers.

### CODE REVIEW

File: lib/puma_auto_tune/master.rb [hunk 1]
```diff
@@ -11,8 +11,8 @@ def running?
     # https://github.com/puma/puma/blob/master/docs/signals.md#puma-signals
     def remove_worker
       previous_worker_count = workers.size
+      send_signal("TTOU")
       until workers.size < previous_worker_count
-        send_signal("TTOU")
```
Comment (schneems) [inline lib/puma_auto_tune/master.rb:15]: This is intentional. We want to be slow to add new workers but as soon as they are added very quick to kill them off until we're under the limit.

File: test/test_helper.rb [hunk 2]
```diff
@@ -17,16 +17,11 @@ def initialize(options = {})
   end
 
   def wait(regex = %r{booted}, timeout = 30)
-    Timeout::timeout(timeout) do
```
Comment (schneems) [inline test/test_helper.rb]: Why do we need to get rid of this?

If you make a mistake coding it's possible to start infinite loops> i wanted some way to get out.

File: lib/puma_auto_tune/master.rb (hunk 1 above)
Comment (penguincoder) [reply lib/puma_auto_tune/master.rb:15]: weird, so this was causing one of the tests to fail. what was happening is that one of the workers was taking too long to go away, so it was killing them repeatedly until there were no workers left. By the time the next callback tick happened, all of the workers were gone and the test couldn't find where it reduced the worker count by one, it reduced it by 5 all at once.

File: test/test_helper.rb (hunk 2 above)
Comment (penguincoder) [reply test/test_helper.rb]: So `Timeout` doesn't work well with threads. I have also seen `Timeout` swallow exceptions and hide what is really happening. There is honestly not a good way to get around the use of Timeout. Ruby 2.3 was throwing an error saying it couldn't find `Timeout`, and I explicitly included it, and it worked. For giggles, I tried removing it and it still worked, so I thought that was a better course.

See: http://www.mikeperham.com/2015/05/08/timeout-rubys-most-dangerous-api/

Comment (schneems) [reply test/test_helper.rb]: I mean you could also link to the article I wrote about this ;)

https://schneems.com/2017/02/21/the-oldest-bug-in-ruby-why-racktimeout-might-hose-your-server/

This is part of the test helpers and it's very much needed.

File: lib/puma_auto_tune/master.rb (hunk 1 above)
Comment (schneems) [reply lib/puma_auto_tune/master.rb:15]: Ahh, interesting. I think that puma has a graceful shutdown time, perhaps we could increase the sleep to match that. Or alternatively maybe we could check to see that the last process we told to die actually exited.

Comment (penguincoder) [reply lib/puma_auto_tune/master.rb:15]: The sleep timer is configurable from what I can recall. I don't think you will ever need to send puma more than the one signal to drop a single process. The sleep is then used to wait until the process goes away.

File: test/test_helper.rb (hunk 2 above)
Comment (penguincoder) [reply test/test_helper.rb]: Well, dang. Okay, I will revert it.

**Pull request closed without merging**